MiniSpace.Services.Email/src/MiniSpace.Services.Email.Core/Entities/EmailContentFactory.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Collections.Generic;
+
+namespace MiniSpace.Services.Email.Core.Entities
+{
+    public class EmailContentFactory
+    {
+        private static readonly Dictionary<NotificationEventType, IEmailContentStrategy> Strategies =
+            new Dictionary<NotificationEventType, IEmailContentStrategy>
+            {
+                { NotificationEventType.TwoFactorCodeGenerated, new TwoFactorCodeGeneratedContent() },
+            };
+
+        public static string CreateContent(NotificationEventType eventType, string details)
+        {
+            if (Strategies.TryGetValue(eventType, out var strategy))
+            {
+                return strategy.GenerateContent(details);
+            }
+            var defaultStrategy = new DefaultEmailContentStrategy();
+            return defaultStrategy.GenerateContent(details);
+        }
+    }
+}
```

Comment:
Nice strategy, although there is not a quite efficient point to use it in the case of the current amount of message content. @SaintAngeLs also, you could use if for the content message in html template mediators factory.